src/Microsoft.DotNet.Build.Tasks/PackageFiles/packageresolve.targets

```diff
@@ -124,18 +124,36 @@
                                        ValidationPatterns="@(ValidationPattern)" />
   </Target>
 
-  <Target Name="FilterTargetingPackResolvedNugetPackages" AfterTargets="ResolveNuGetPackages"
-          Condition="'$(TargetingPackNugetPackageId)' != ''">
+  <Target Name="FilterTargetingPackResolvedNugetPackages" AfterTargets="ResolveNuGetPackages" >
+    <PropertyGroup>
+      <_TargetingPackPrefix>Microsoft.TargetingPack</_TargetingPackPrefix>
+    </PropertyGroup>
+
+    <!--
+      Add the core libraries and windows to the reference set by default to avoid a lot of duplication in projects
+      They only act as a filter so if they aren't present in the packages references it will not impact anything.
+    -->
+    <ItemGroup Condition="'$(ExcludeDefaultTargetingPackReferences)' != 'true'">
+      <TargetingPackReference Include="mscorlib" />
+      <TargetingPackReference Include="System.Private.CoreLib" />
```

Comment:
Are we sure we want to default include this one? Most of the assemblies in the .NET Native targeting pack are actually contract-based, and a lot of the .NET Native facades use the targeting pack without referencing System.Private.CoreLib. See System.Reflection.Extensions, System.IO, System.Resources.ResourceManager, others. I would guess that those stop building after this change unless we added the Exclude property to them.

I think it's okay to include Windows.winmd because it's in it's own targeting pack.